src/UnitTests/GitHub.App/Models/AccountModelTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading;
+using System.Windows.Media.Imaging;
+using GitHub.Collections;
+using GitHub.Models;
+using GitHub.Services;
+using ReactiveUI;
+using Xunit;
+
+namespace UnitTests.GitHub.App.Models
+{
+    public class AccountModelTests : TestBaseClass
+    {
+        [Fact]
+        public void CopyFromDoesNotLoseAvatar()
+        {
+            //A function that will return this image in an observable after X seconds
+            var userImage = AvatarProvider.CreateBitmapImage("pack://application:,,,/GitHub.App;component/Images/default_user_avatar.png");
+            var orgImage = AvatarProvider.CreateBitmapImage("pack://application:,,,/GitHub.App;component/Images/default_org_avatar.png");
+
+            Func<int, BitmapImage, IObservable<BitmapImage>> generateObservable = (seconds, image) =>
+            {
+                return Observable.Generate(
+                    initialState: 0, 
+                    condition: i => i < 1, 
+                    iterate: state => state + 1,
+                    resultSelector: i => image, 
+                    timeSelector: i => TimeSpan.FromSeconds(seconds));
+            };
```

Comment:
Like @grokys mentioned in the other PR, timings are iffy, it's safer to use a `Subject` or perhaps a `ReplaySubject` and do `OnNext` on it whenever you want to push something. Saves on waits, too :)